crates/typst-kit/src/package.rs

```diff
@@ -186,3 +203,80 @@ impl PackageStorage {
         })
     }
 }
+
+/// An iterator that deserializes its items lazily.
+struct LazyDeser<T, I> {
```

Comment:
Maybe this struct and related code could go in their own module?

Reply:
I think it's okay to leave it here until we need it somewhere else